### PR TITLE
Bigtable Admin: Promote models to top level classes

### PR DESCRIPTION
**Pull request by igorbernstein2**
This aims be a mechanical structural change. It promotes the inner classes of `TableAdminRequests` & `TableAdminResponses` to be toplevel classes in `com.google.cloud.bigtable.admin.v2.models` and moves all of the static `convertX` methods in `TableAdminResponses` to be static `fromProto` methods in the relevant model.

### CODE REVIEW

**Comment by igorbernstein2** [comment]
@kevinsi4508 can you take a look?

**Comment by kevinsi4508** [comment]
Sure, taking a look.

File: google-cloud-clients/google-cloud-bigtable-admin/src/main/java/com/google/cloud/bigtable/admin/v2/models/ModifyColumnFamiliesRequest.java [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+  /**
+   * Configures the name and GcRule of the new ColumnFamily to be created
+   *
+   * @param familyId
+   * @param gcRule
+   * @return
+   */
+  public ModifyColumnFamiliesRequest addFamily(String familyId, GCRule gcRule) {
+    Modification.Builder modification = Modification.newBuilder().setId(familyId);
+    Preconditions.checkNotNull(gcRule);
```
Comment (kevinsi4508) [inline google-cloud-clients/google-cloud-bigtable-admin/src/main/java/com/google/cloud/bigtable/admin/v2/models/ModifyColumnFamiliesRequest.java]: Move checkNotNull up.

File: google-cloud-clients/google-cloud-bigtable-admin/src/main/java/com/google/cloud/bigtable/admin/v2/models/ModifyColumnFamiliesRequest.java [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+    modification.setCreate(ColumnFamily.newBuilder().setGcRule(gcRule.toProto()));
```
Comment (kevinsi4508) [inline google-cloud-clients/google-cloud-bigtable-admin/src/main/java/com/google/cloud/bigtable/admin/v2/models/ModifyColumnFamiliesRequest.java]: modification.getCreateBuilder().setGcRule()

File: google-cloud-clients/google-cloud-bigtable-admin/src/main/java/com/google/cloud/bigtable/admin/v2/models/ModifyColumnFamiliesRequest.java [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+
+  /**
+   * Drops the specified ColumnFamily
+   *
+   * @param familyId
+   * @return
+   */
+  public ModifyColumnFamiliesRequest dropFamily(String familyId) {
+    Modification.Builder modification = Modification.newBuilder().setId(familyId);
+    modification.setId(familyId).setDrop(true);
```
Comment (kevinsi4508) [inline google-cloud-clients/google-cloud-bigtable-admin/src/main/java/com/google/cloud/bigtable/admin/v2/models/ModifyColumnFamiliesRequest.java]: .newBuilder()
.setId()
.setDrop()

File: google-cloud-clients/google-cloud-bigtable-admin/src/main/java/com/google/cloud/bigtable/admin/v2/models/Table.java [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+import com.google.common.collect.ImmutableMap;
+import java.util.Collection;
+import java.util.Map;
+import java.util.Map.Entry;
+import javax.annotation.Nonnull;
+
+/** Wrapper for {@link Table} protocol buffer object */
+public final class Table {
+  private final TableName tableName;
+  private final Map<String, ClusterState> clusterStates;
```
Comment (kevinsi4508) [inline google-cloud-clients/google-cloud-bigtable-admin/src/main/java/com/google/cloud/bigtable/admin/v2/models/Table.java]: Document what are the keys?

Both ClusterState and ColumnFamily have an ID field.

Are the keys and IDs the same? Why do we need both?

File: google-cloud-clients/google-cloud-bigtable-admin/src/main/java/com/google/cloud/bigtable/admin/v2/models/Table.java [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+
+  @InternalApi
+  public static Table fromProto(@Nonnull com.google.bigtable.admin.v2.Table proto) {
+    ImmutableMap.Builder<String, ClusterState> clusterStates = ImmutableMap.builder();
+
+    for (Entry<String, com.google.bigtable.admin.v2.Table.ClusterState> entry : proto.getClusterStatesMap().entrySet()) {
+      clusterStates.put(entry.getKey(), ClusterState.fromProto(entry.getKey(), entry.getValue()));
+    }
+
+    ImmutableMap.Builder<String, ColumnFamily> families = ImmutableMap.builder();
```
Comment (kevinsi4508) [inline google-cloud-clients/google-cloud-bigtable-admin/src/main/java/com/google/cloud/bigtable/admin/v2/models/Table.java]: "families" or "columnFamilies", be consistent.

File: google-cloud-clients/google-cloud-bigtable-admin/src/main/java/com/google/cloud/bigtable/admin/v2/models/Table.java [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+   * Returns a map of columfamilies in the table keyed by columnfamily and name
+   */
+  public Map<String, ColumnFamily> getColumnFamiliesMap() {
+    return columnFamilies;
+  }
+
+  /**
+   * Returns state of the table by clusters in the instance as a Collection
+   */
+  public Collection<ClusterState> getClusterStates() {
```
Comment (kevinsi4508) [inline google-cloud-clients/google-cloud-bigtable-admin/src/main/java/com/google/cloud/bigtable/admin/v2/models/Table.java]: What do we need getClusterStatesMap() and getClusterStates()?

File: google-cloud-clients/google-cloud-bigtable-admin/src/main/java/com/google/cloud/bigtable/admin/v2/models/Table.java [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+
+  @Override
+  public String toString() {
+    return MoreObjects.toStringHelper(this)
+        .add("tableName", tableName)
+        .add("clusterStates", getClusterStates())
+        .add("columnFamiles", getColumnFamiles())
+        .toString();
+  }
+
```
Comment (kevinsi4508) [inline google-cloud-clients/google-cloud-bigtable-admin/src/main/java/com/google/cloud/bigtable/admin/v2/models/Table.java]: nit: delete extra lines

File: google-cloud-clients/google-cloud-bigtable-admin/src/test/java/com/google/cloud/bigtable/admin/v2/models/ConsistencyTokenTest.java [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+public class ConsistencyTokenTest {
+  @Test
+  public void fromJsonTest() {
+    ConsistencyToken tokenResponse =
+        ConsistencyToken.fromProto(
+            GenerateConsistencyTokenResponse.newBuilder()
+                .setConsistencyToken("87282hgwd8yg")
+                .build());
+
+    assertNotNull(tokenResponse);
```
Comment (kevinsi4508) [inline google-cloud-clients/google-cloud-bigtable-admin/src/test/java/com/google/cloud/bigtable/admin/v2/models/ConsistencyTokenTest.java]: delete

**Comment by igorbernstein2** [comment]
Thank you for reviewing.  All feedback should either be addressed. Please note that I added TODOs for most of the comments because these are pre-existing issues that I plan on addressing in followup PRs and this PR is pretty noisy as is

**Comment by kevinsi4508** [comment]
LGTM

**Comment by igorbernstein2** [comment]
All feedback addressed. Thanks for reviewing.

@garrettjonesgoogle would mind taking a look?

**Pull request merged**